### PR TITLE
Suggestion Feature for Issue#655

### LINKED ISSUE #655: Suggestions/feedback plugin
Create a plugin that will allow for a user to submit an anonymous suggestion for the server. The Moderation/Admin team should ideally review the submitted suggestions once a month or cycle and implement suggestions at their discretion. Suggestions can be displayed in a read-only channel.

Invocation could be something like:
`!suggest <suggestion>`

ex. !suggest Creating an NFT channel.

### PR DESCRIPTION
**Pull request by bquach2**
Implemented Suggestion plugin requested in issue #655 and created a test case for it. Fixes #655

### CODE REVIEW

File: src/app/plugins/suggestions.plugin.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { MessageEmbed} from 'discord.js';
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType } from '../../common/types';
+
+export default class SuggestionPlugin extends Plugin {
+  public commandName: string = 'suggest';
+  public name: string = 'Suggestions Plugin';
+  public description: string = 'Create suggestions for lion to output';
+  public usage: string = 'suggest <string>';
```
Comment (c650) [inline src/app/plugins/suggestions.plugin.ts:9]: Only suggestions channel

File: src/app/plugins/suggestions.plugin.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  public permission: ChannelType = ChannelType.All;
+  private _embed: MessageEmbed = new MessageEmbed();
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public async execute(message: IMessage, args: string[]) {
```
Comment (c650) [inline src/app/plugins/suggestions.plugin.ts:16]: Can we rate limit to prevent abuse

**Review by c650 — CHANGES_REQUESTED** [review]

**Comment by c650** [comment]
I think this is good for #672 too/instead

**Comment by c650** [comment]
I think this is good for #672 too/instead

**Comment by bquach2** [comment]
Updated to only work in Suggestions channel, thanks!

File: src/app/plugins/suggestions.plugin.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import Constants from '../../common/constants';
+
+export default class SuggestionPlugin extends Plugin {
+  public commandName: string = 'suggest';
+  public name: string = 'Suggestions Plugin';
+  public description: string = 'Create suggestions for lion to output';
+  public usage: string = 'suggest <string>';
+  public override pluginChannelName: string = Constants.Channels.Public.ServerRequests;
+  public permission: ChannelType = ChannelType.All;
+  private _embed: MessageEmbed = new MessageEmbed();
```
Comment (tanndlin) [inline src/app/plugins/suggestions.plugin.ts:13]: I dont see a reason to store the embed as a property of the class when nothing is carried over from each use.

You could add a footer to the embed to justify this, but even then `new MessageEmbed` is not an expensive operation.
I would just make a new one each time

File: src/app/plugins/suggestions.plugin.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const suggestion = args.join(' ');
+    // Combines input
+    this._embed.setTitle('Suggestion:');
+    this._embed.setDescription(suggestion);
+    await message.reply({ embeds: [this._embed] });
```
Comment (tanndlin) [inline src/app/plugins/suggestions.plugin.ts:23]: Would be nice to have Lion react with a thumbs up then thumbs to down
to imply to the user that you should react to vote

**Review by tanndlin — CHANGES_REQUESTED** [review]

**Comment by bquach2** [comment]
I updated to make a new embed object every time, thanks!

**Pull request closed without merging**